### PR TITLE
corp: support in push down

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @coocood @qiuyesuifeng PTAL

### CODE REVIEW

**Comment by ngaut** [comment]
PTAL @coocood

File: src/util/xeval/evaluator.rs [hunk 1]
```diff
@@ -183,12 +187,44 @@ impl Evaluator {
...
+        }
+        if decoded.first().map_or(false, |d| *d == Datum::Null) {
+            return Ok(Datum::Null);
+        }
+        Ok(false.into())
+    }
+
+    fn decode_value_list(&mut self, value_list_expr: &Expr) -> Result<&Vec<Datum>> {
+        let p = value_list_expr as *const Expr as isize;
+        let encoded = try!(self.cached_value_list
```
Comment (coocood) [inline src/util/xeval/evaluator.rs]: encoded -> decoded

**Comment by coocood** [comment]
Rest LGTM

**Comment by BusyJay** [comment]
@ngaut @qiuyesuifeng PTAL

**Comment by ngaut** [comment]
LGTM

**Pull request merged**